### PR TITLE
Add http status code based authentication for rtmp and hls

### PR DESCRIPTION
**Pull request by melvinheks**
Utilizes a localhost only location that passes along the query
arugments of a rtmp url connection through a post request to the
specified url.
Ex : rtmp://ip:port/show/id?var1=value1&var2=value2
var1=value1 and var2=value2 will be sent as a post request to the
url in the proxy_pass directive of api-https-proxy

Remake of https://github.com/meowl-surveillance-system/meowl/pull/31

### CODE REVIEW

File: scripts/lib/install_nginx_rtmp.sh [hunk 1]
```diff
@@ -85,7 +95,25 @@ http {
...
+    server {
+        listen 80;
+        server_name localhost;
+        location /api-https-proxy {
+          allow 127.0.0.1;
+          deny all;
+          # Strips the api-https-proxy part of
+          # the url, but forward the rest.
+          rewrite  ^/api-https-proxy/(.*) /\$1 break;
+          proxy_pass https://example.com/login/;
```
Comment (melvinheks) [inline scripts/lib/install_nginx_rtmp.sh]: /auth/rtmpAuth

Comment (yliu0715) [reply scripts/lib/install_nginx_rtmp.sh]: yes

File: src/lib/nginx-rtmp/install [hunk 2]
```diff
@@ -21,15 +21,16 @@ STREAMING_SERVER_KEY="${KEY_PATH}/meowl_nginx.key"
 STREAMING_SERVER_CERT="${CERT_PATH}/meowl_nginx.crt"
 STREAMING_SERVER_IP=127.0.0.1
 STREAMING_SERVER_DNS="localhost"
+AUTH_SERVER_URL="https://example.com"
```
Comment (chc5) [inline src/lib/nginx-rtmp/install:24]: what is this

Comment (melvinheks) [reply src/lib/nginx-rtmp/install:24]: Authorization server / react express website

**Pull request merged**